### DISCUSSION TITLE
Train layout automation

### DISCUSSION THREAD

**Discussion by Nelvarion** [pybricks/support#227] [Show and tell]
This is an adaptation of the hub->pc->hub communication demo.
The purpose is for a hub with motors connected to 1 or more switch points to change the track direction based on input from an oncoming train(s).
In addition, it will include other cases for the trains - stopping at stations, slowing/speeding, etc. - based on color sensor mounted on the trains, using colored tiles on the track (custom colors are defined to increase the possible cases).
Main change from the original notebook is wrapping the connection to the different hubs in `try catch` to allow each component to work independently from the others - this will let you stop each piece for battery replacement or maintenance, without having to reset the entire layout.
It's still a work in progress:
1. SW
1.1 Add the different cases for trains
1.2 Add a remote control to allow for an interactive layout in shows - let people control different parts of the layout
2. HW
2.1 Utilize Control+ hub(s) for switch control (once my BL order finally arrives)
2.2 Build extension cables to reduce the number of hubs needed and increase the reach of each hub

Any feedback or comments are welcome!
https://github.com/Nelvarion/pybricksdev-demo/blob/master/test-wrapper.ipynb

**Comment by laurensvalk** [comment]
This is great, thanks for sharing! Is there a video of this system in action somewhere?

**Comment by Nelvarion** [comment]
Not yet, coding is silent, LEGO trains... not so much. 😅
Should have a video this week.

**Comment by Nelvarion** [comment]
As promised, a quick video demonstrating the project.
https://youtu.be/ZJDu5gDbE54

**Reply by laurensvalk** [reply]
I like it! Nice work.
